scripts/layouts_tool/layoutstoolmain.cpp

```diff
@@ -0,0 +1,224 @@
+#include "layoutstoolmain.h"
+#include "utils.h"
+#include <QDebug>
+
+LayoutsToolMain::LayoutsToolMain(QObject *parent) :
+        QObject(parent) {
+
+    // Find layouts.h path (check for Mixxx directory four directories up)
+    bool layoutsFound = false;
+    QDir cp = QDir::currentPath();
+
+    for (int i = 0; i < 5; i++) {
+        QString path(cp.absolutePath()+"/src/controllers/keyboard/layouts.cpp");
+        QFileInfo checkFile(path);
+
+        if (checkFile.exists() && checkFile.isFile()) {
+            mFilePath = path;
+            qDebug() << "Found path: " << path;
+            layoutsFound = true;
+        }
+        cp.cdUp();
+    }
+
+    if (!layoutsFound) {
+        mFilePath = "";
+    }
+
+    app = qApp;
+    pLayoutsFileHandler = new LayoutsFileHandler();
+}
+
+void LayoutsToolMain::run() {
+    qDebug() << "Welcome to the Layouts tool :)";
```

Comment:
Done! I also made a similar function for `QTextStream(stdin)` which makes all a lot more clear. Thanks for pointing this out!